src/richie/apps/courses/migrations/0028_auto_20210126_2200.py

```diff
@@ -0,0 +1,35 @@
+# Generated by Django 3.1.5 on 2021-01-26 21:00
+
+from django.db import migrations
+
+
+def delete_orphan_course_run_nodes(apps, schema_editor):
+    """
+    When we deleted the pages previously related to course runs, we
+    forgot to delete the related nodes. Let's fix this.
+    """
+    TreeNode = apps.get_model("cms", "TreeNode")
+
+    for node in (
+        TreeNode.objects.filter(
+            cms_pages__isnull=True, parent__cms_pages__course__isnull=False
+        )
+        .distinct()
+        .iterator()
+    ):
+        # When deleting a node, we must also update its parent to decrement
+        # its number of children
+        parentpath = node.path[0 : (node.depth - 1) * 4]
+        parent = TreeNode.objects.get(path=parentpath)
+        node.delete()
+        parent.numchild -= 1
+        parent.save()
```

Comment:
Treebeard documentation advises to consider `num_child` as read-only.
After node deletion, `fix_tree` should do the job to detect and fix inconsistency in parent node.

```suggestion
        parent = node.get_parent()
        node.delete()
        parent.fix_tree()
        parent.save()
```

https://django-treebeard.readthedocs.io/en/latest/mp_tree.html#treebeard.mp_tree.MP_Node.fix_tree

Reply:
I know because I went through the whole process of "why just deleting the node doesn't fix my tree" :smile: I was just trying to make it fast because running `fix_tree` takes forever and I didn't want to run it in a migration. They say it's read-only because they are keeping it up-to-date but in my case they don't because a migration is operating on a dummy model that does not have any of the model's methods... 
What do you think? Should I call the "fix_tree" command in my migration?

Reply:
Ok, if `fix_tree` is a heavy operation, let's keep your implementation.